spec.md

```diff
@@ -541,6 +543,10 @@ The tags MUST be in lexical order.
 
 When using the `last` query parameter, the `n` parameter is OPTIONAL.
 
+The registry SHOULD support at least `n=100` tags.
+The client SHOULD avoid requesting more than `n=100` tags.
+If the registry is unable to return all tags requested by the client, the registry MUST return a `400 Bad Request` with `TAG_LIMIT_EXCEEDED` error <sup>[code-15](#error-codes)</sup>.
```

Comment:
Maybe it would make sense to require the maximum number of tags to be included in the response in this case so that clients don't have to guess the maximum number (and potentially issue more unservable requests).

Reply:
Makes sense. The goal with the `n=100` was to give clients a number they know should be a safe default. Do you believe registries will try to go lower than that? Or is this planning for clients that try to pull 1000 tags and the registry bumps them down to 250?

Should this be done with an OCI header in the error response?

Reply:
Honestly I wouldn't know why registries should go lower than 100 as tag lookup shouldn't be a costly operation, anyway. But on the client side I would expect clients to try to go over that to compile a list of all tags (e.g. for filtering/searching purposes). In that case it would incur less complexity on the client side (implement an algorithm for finding the lower barrier) when the registry would just say "nope, 1000 is too much, but 250 is good" and the client would only issue 2 requests in total to get to that.

About how to return that limit I'm unsure as the "Error Codes" section is very open with regards to what registries should return, e.g. registries may choose to not return a body at all so a mandate to set an OCI header in the response sounds reasonable. IIRC there is no precedence for requirements of a certain header to present in 4xx responses so this would be kind of an outlier but IMHO a well justified one.

OCI-Tag-Limit?